bocadillo/applications.py

```diff
@@ -462,30 +453,16 @@ def run(
         if port is None:
             port = 8000
 
-        if not debug:
-            _run(self, host=host, port=port, **kwargs)
-            return
-
-        self.debug = True
-
-        # Reload static files in development.
-        # See: http://whitenoise.evans.io/en/stable/base.html#autorefresh
-        for whitenoise in self._static_apps.values():
-            whitenoise.autorefresh = True
-
-        if live_server:
-            kwargs = {
-                "app": self,
-                "host": host,
-                "port": port,
-                "log_level": log_level,
-                "debug": self.debug,
-                **kwargs,
-            }
-            reloader = StatReload(get_logger(log_level))
-            reloader.run(_run, kwargs)
-        else:
-            _run(self, host=host, port=port, **kwargs)
+        if debug:
+            self.debug = True
+            kwargs["reload"] = True
```

Comment:
Stick with just using `kwargs["debug"] = True` here is the simplest thing for now, since you're currently relying on the debug middleware.

You'll want to use debug middleware explicitly when we move to 0.6, but I can issue a PR showing how to do that - and the code is properly pinned against `0.5.*` at this point so you won't see any breaking changes.

Reply:
Yep, sounds right. I forgot that the handling of uncaught exceptions to 500 errors was done at the uvicorn level. :-)